### PR TITLE
fix(lighthouse): get correct report URL

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

- Fix Lighthouse report URLs.
- Bump lighthouse version to v0.10

## Why?

Recently, the URLs have stopped pointing to reports, which makes it harder to investigate them.

Now we get proper storage URLs again: https://storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/1676462117644-93263.report.html

### CODE REVIEW

File: dotcom-rendering/src/web/components/BigNumber/BigNumber.tsx [hunk 1]
```diff
@@ -10,6 +9,12 @@ import Number7 from './7.svg';
 import Number8 from './8.svg';
 import Number9 from './9.svg';
 
+const Number0 = () => (
+	<svg width="34" height="40">
+		<path d="M17 40C7.8 40 0 33.2 0 20.2 0 6.6 8.1 0 17.3 0 26.4 0 34 6.5 34 19.9S26.3 40 17 40zm.1-1c4 0 5.2-6.1 5.2-18.7C22.3 7 21 1.1 17.1 1.1c-4 0-5.4 6-5.4 18.7 0 13 1.4 19.1 5.4 19.2z" />
+	</svg>
+);
+
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/BigNumber/BigNumber.tsx:17]: This was done to force a Lighthouse report to be generated–will remove.

File: scripts/deno/surface-lighthouse-results.ts [hunk 2]
```diff
@@ -42,32 +43,25 @@ console.log(`Using issue #${issue_number}`);
...
-	values: number[];
-	passed: boolean;
-	level?: LHCI.AssertCommand.AssertionFailureLevel;
-	auditId?: string;
-	auditProperty?: string;
-	auditTitle?: string;
-	auditDocumentationLink?: string;
-	message?: string;
-}
+type AssertionResult = LHCI.AssertResults.AssertionResult;
```
Comment (bryophyta) [inline scripts/deno/surface-lighthouse-results.ts:46]: 🥳

File: scripts/deno/surface-lighthouse-results.ts [hunk 3]
```diff
@@ -42,32 +43,25 @@ console.log(`Using issue #${issue_number}`);
...
 const results: AssertionResult[] = JSON.parse(
-	Deno.readTextFileSync(`${dir}/assertion-results.json`),
+	await Deno.readTextFile(`${dir}/assertion-results.json`),
 );
 
+const testUrl = new URL(Deno.env.get('LHCI_URL') ?? 'http://localhost:9000/');
```
Comment (bryophyta) [inline scripts/deno/surface-lighthouse-results.ts:51]: great to use a URL object here!

**Review by bryophyta — APPROVED** [review]
I haven't run the script but assuming it's been tested the code looks good to me! Some nice refactors as well :)

**Pull request merged**